plugins/unifying/fu-plugin-unifying.c

```diff
@@ -10,205 +10,163 @@
 }
 
 gboolean
-fu_plugin_update_detach (FuPlugin *plugin, FuDevice *dev, GError **error)
+fu_plugin_update (FuPlugin *plugin,
+		  FuDevice *device,
+		  GBytes *blob_fw,
+		  FwupdInstallFlags flags,
+		  GError **error)
 {
-	FuPluginData *data = fu_plugin_get_data (plugin);
-	LuDevice *device = LU_DEVICE (dev);
-
-	/* get device */
-	if (!lu_device_open (device, error))
+	g_autoptr(FuDeviceLocker) locker = fu_device_locker_new (device, error);
+	if (locker == NULL)
 		return FALSE;
+	return fu_device_write_firmware (device, blob_fw, error);
+}
 
-	/* switch to bootloader if required */
-	if (!lu_device_has_flag (device, LU_DEVICE_FLAG_REQUIRES_DETACH))
-		return TRUE;
-
-	/* wait for device to come back */
-	fu_device_set_status (dev, FWUPD_STATUS_DEVICE_RESTART);
-	if (lu_device_has_flag (device, LU_DEVICE_FLAG_DETACH_WILL_REPLUG)) {
-		g_debug ("doing detach in idle");
-		g_idle_add_full (G_PRIORITY_DEFAULT_IDLE,
-				 fu_plugin_unifying_detach_cb,
-				 g_object_ref (dev),
-				 (GDestroyNotify) g_object_unref);
-		if (!lu_context_wait_for_replug (data->ctx,
-						 device,
-						 FU_DEVICE_REMOVE_DELAY_RE_ENUMERATE,
-						 error))
-			return FALSE;
-	} else {
-		g_debug ("doing detach in main thread");
-		if (!fu_device_detach (dev, error))
-			return FALSE;
+static gboolean
+fu_plugin_unifying_check_supported_device (FuPlugin *plugin, FuDevice *device)
+{
+	GPtrArray *guids = fu_device_get_guids (device);
+	for (guint i = 0; i < guids->len; i++) {
+		const gchar *guid = g_ptr_array_index (guids, i);
+		if (fu_plugin_check_supported (plugin, guid))
+			return TRUE;
 	}
-	return TRUE;
+	return FALSE;
 }
 
 gboolean
-fu_plugin_update_attach (FuPlugin *plugin, FuDevice *dev, GError **error)
+fu_plugin_udev_device_added (FuPlugin *plugin, FuUdevDevice *device, GError **error)
 {
-	FuPluginData *data = fu_plugin_get_data (plugin);
-	LuDevice *device = LU_DEVICE (dev);
+	g_autoptr(FuDevice) dev = NULL;
+	g_autoptr(FuDeviceLocker) locker = NULL;
 
-	/* get device */
-	if (!lu_device_open (device, error))
-		return FALSE;
+	/* interesting device? */
+	if (g_strcmp0 (fu_udev_device_get_subsystem (device), "hidraw") != 0)
+		return TRUE;
 
-	/* wait for it to appear back in runtime mode if required */
-	if (!lu_device_has_flag (device, LU_DEVICE_FLAG_REQUIRES_ATTACH))
+	/* logitech */
+	if (fu_udev_device_get_vendor (device) != FU_UNIFYING_DEVICE_VID)
 		return TRUE;
 
-	/* wait for device to come back */
-	fu_device_set_status (dev, FWUPD_STATUS_DEVICE_RESTART);
-	if (lu_device_has_flag (device, LU_DEVICE_FLAG_ATTACH_WILL_REPLUG)) {
-		g_debug ("doing attach in idle");
-		g_idle_add_full (G_PRIORITY_DEFAULT_IDLE,
-				 fu_plugin_unifying_attach_cb,
-				 g_object_ref (device),
-				 (GDestroyNotify) g_object_unref);
-		if (!lu_context_wait_for_replug (data->ctx,
-						 device,
-						 FU_DEVICE_REMOVE_DELAY_RE_ENUMERATE,
-						 error))
-			return FALSE;
+	/* runtime */
+	if (fu_device_has_custom_flag (FU_DEVICE (device), "is-runtime")) {
```

Comment:
Do you need a custom flag here really?  Can you just look for the absence of `FWUPD_DEVICE_FLAG_IS_BOOTLOADER` instead?